full-service/src/service/sync.rs

```diff
@@ -291,6 +291,18 @@ fn sync_thread_entry_point(
                     // Errors that are acceptable - nothing to do.
                     Err(SyncError::AccountNotFound) => {}
 
+                    Err(SyncError::Database(WalletDbError::NoDecryptionKey)) => {
+                        log::trace!(
+                            logger,
+                            "No decryption key. Cannot sync account {:?} Please unlock database.",
+                            account_id,
+                        );
+                        // Queue the account again.
+                        sender
```

Comment:
Wouldn't this result in 100% cpu usage?

Reply:
Ah yes - I wanted to make sure the account was queued again so that it would sync when the DB was unlocked, but the account should get picked up in the next iteration of the SyncThread's start loop. I removed this queue step and and verified that the account gets picked up.